prov/rxd/src/rxd_cq.c

```diff
@@ -286,15 +285,26 @@ void rxd_post_cts(struct rxd_ep *rxd_ep, struct rxd_x_entry *rx_entry,
 		rxd_release_tx_pkt(rxd_ep, pkt_entry);
 }
 
-static int rxd_match_recv(struct dlist_entry *item, const void *arg)
+static int rxd_match_msg(struct dlist_entry *item, const void *arg)
 {
-	const struct rxd_pkt_entry *pkt_entry = arg;
+	const struct rxd_pkt_entry *pkt_entry = (struct rxd_pkt_entry *) arg;
 	struct rxd_x_entry *rx_entry;
 
 	rx_entry = container_of(item, struct rxd_x_entry, entry);
 
-	return (rx_entry->peer == FI_ADDR_UNSPEC ||
-		rx_entry->peer == pkt_entry->peer);
+	return rxd_match_addr(rx_entry->peer, pkt_entry->peer);
+}
+
+static int rxd_match_tagged(struct dlist_entry *item, const void *arg)
```

Comment:
nit: rename to rxd_match_tmsg() -- to avoid confusion with rxd_match_tag() function.